Snowflake.Data/Core/ArrowResultSet.cs

```diff
@@ -0,0 +1,168 @@
+                    {
+                        var recordBatch = reader.ReadNextRecordBatch();
+                        _currentChunkRowCount = recordBatch.Length;
+                        _currentChunk = new ArrowResultChunk(recordBatch);
+                    }
+                }
+
+                this.sfStatement = sfStatement;
+                UpdateSessionStatus(responseData);
+
+                if (responseData.chunks != null)
+                {
+                    _totalChunkCount = responseData.chunks.Count;
+                    
+                    // TODO in SNOW-893835 - support for multiple chunks
+                    throw new SnowflakeDbException(SFError.UNSUPPORTED_FEATURE);
+                }
+
+                responseData.rowSet = null;
+
+                sfResultSetMetaData = new SFResultSetMetaData(responseData);
+
+                isClosed = false;
+
+                queryId = responseData.queryId;
+            }
+            catch(System.Exception ex)
+            {
+                s_logger.Error("Result set error queryId="+responseData.queryId, ex);
+                throw;
+            }
+        }
+
+        internal override async Task<bool> NextAsync()
+        {
+            ThrowIfClosed();
+
+            _currentChunkRowIdx++;
+            if (_currentChunkRowIdx < _currentChunkRowCount)
+            {
+                return true;
+            }
+
+            if (_totalChunkCount > 0)
+            {
+                // TODO in SNOW-893835 - support for multiple chunks
+                throw new SnowflakeDbException(SFError.UNSUPPORTED_FEATURE);
+            }
+
+            return false;
+        }
+
+        internal override bool Next()
+        {
+            ThrowIfClosed();
+
+            _currentChunkRowIdx++;
+            if (_currentChunkRowIdx < _currentChunkRowCount)
+            {
+                return true;
+            }
+
+            if (_totalChunkCount > 0)
+            {
+                // TODO in SNOW-893835 - support for multiple chunks
+                throw new SnowflakeDbException(SFError.UNSUPPORTED_FEATURE);
+            }
+
+            return false;
+        }
+
+        internal override bool NextResult()
+        {
+            return false;
+        }
+
+        internal override async Task<bool> NextResultAsync(CancellationToken cancellationToken)
+        {
+            return await Task.FromResult(false);
+        }
+
+        internal override bool HasRows()
+        {
+            if (isClosed)
+            {
+                return false;
+            }
+
+            return _currentChunkRowCount > 0 || _totalChunkCount > 0;
+        }
+
+        /// <summary>
+        /// Move cursor back one row.
+        /// </summary>
+        /// <returns>True if it works, false otherwise.</returns>
+        internal override bool Rewind()
+        {
+            ThrowIfClosed();
+
+            if (_currentChunkRowIdx >= 0)
+            {
+                _currentChunkRowIdx--; 
```

Comment:
is it doable (in the SNOW-893835) to rewind to a row from previous chunk? If yes I think we lack a comment TODO here